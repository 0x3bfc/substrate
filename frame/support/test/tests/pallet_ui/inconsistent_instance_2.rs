#[frame_support::pallet(Example)]
mod pallet {
	use frame_support::pallet_prelude::ModuleInterface;
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::trait_]
	pub trait Trait: frame_system::Trait {}

	#[pallet::module]
	pub struct Module<T, I = DefaultInstance>(core::marker::PhantomData<(T, I)>);

	#[pallet::module_interface]
	impl<T: Trait<I>, I: Instance> ModuleInterface<BlockNumberFor<T>> for Module<T, I> {}

	#[pallet::call]
	impl<T: Trait<I>, I: Instance> Module<T, I> {}
}

fn main() {
}
