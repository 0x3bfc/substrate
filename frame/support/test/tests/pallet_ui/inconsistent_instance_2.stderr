error: Invalid generic declaration, trait is defined without instance but generic use some
  --> $DIR/inconsistent_instance_2.rs:16:7
   |
16 |     impl<T: Trait<I>, I: Instance> Module<T, I> {}
   |          ^

error: Invalid generic declaration, trait is defined without instance but generic use some
  --> $DIR/inconsistent_instance_2.rs:16:33
   |
16 |     impl<T: Trait<I>, I: Instance> Module<T, I> {}
   |                                    ^^^^^^

error: Invalid generic declaration, trait is defined without instance but generic use some
  --> $DIR/inconsistent_instance_2.rs:10:20
   |
10 |     pub struct Module<T, I = DefaultInstance>(core::marker::PhantomData<(T, I)>);
   |                       ^

error: Invalid generic declaration, trait is defined without instance but generic use some
  --> $DIR/inconsistent_instance_2.rs:13:72
   |
13 |     impl<T: Trait<I>, I: Instance> ModuleInterface<BlockNumberFor<T>> for Module<T, I> {}
   |                                                                           ^^^^^^

error: Invalid generic declaration, trait is defined without instance but generic use some
  --> $DIR/inconsistent_instance_2.rs:13:7
   |
13 |     impl<T: Trait<I>, I: Instance> ModuleInterface<BlockNumberFor<T>> for Module<T, I> {}
   |          ^
